src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,282 @@
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+
+/**
+ * Parser deals with making sense of the user command.
+ */
+public class Parser {
+    private String commandLine;
+
+    /**
+     * Parser constructor.
+     * @param commandLine The string of text given by the user.
+     */
+    Parser(String commandLine) {
+        this.commandLine = commandLine;
+    }
+
+    /**
+     * The main method that analyse the user input and give an
+     * appropriate command.
+     * @param commandLine The user input.
+     * @return The command based on the user input. The 5 commands are AddCommand, ListCommand,
+     * DeleteCommand, DoneCommand and ExitCommand.
+     * @throws DukeException Throws exceptions that are all sub class of the DukeException.
+     */
+    public static Command parse(String commandLine) throws DukeException {
+        try {
+            if (isList(commandLine)) {
+                return parseListCommand();
+            } else if (isToDo(commandLine)) {
+                return parseAddCommandTodo(commandLine);
+            } else if (isDeadline(commandLine) || isEvent(commandLine)) {
+                return parseAddCommandWithDate(commandLine);
+            } else if (isDone(commandLine) || isDelete(commandLine)) {
+                return parseDoneDeleteCommand(commandLine);
+            } else if (isFind(commandLine)) {
+                return parseFindCommand(commandLine);
+            } else if (commandLine.equals("bye")) {
+                return new ExitCommand();
+            } else {
+                throw new DukeInvalidTaskException();
+            }
+        } catch (DateTimeException e) {
+            throw new DukeDateTimeParseException();
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new DukeInvalidIndexException();
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            throw new DukeEmptyActionException();
+        }
+    }
+
+    /**
+     * Parses the keyword from the commandline and use it as an argument for
+     * creating the FindCommand.
+     * @param commandLine The user input.
+     * @return FindCommand.
+     * @throws DukeEmptyFindException Checks if the keyword is empty.
+     */
+    private static Command parseFindCommand(String commandLine) throws DukeEmptyFindException {
+        if (commandLine.equals("todo")) {
+            throw new DukeEmptyFindException();
+        }
+        return new FindCommand(commandLine.substring(5));
+    }
+
+    /**
+     * Check if the input starts with 'find'.
+     * @param commandLine
+     * @return
+     */
+    private static boolean isFind(String commandLine) {
+        return commandLine.startsWith("find");
+    }
+
+    /**
+     * Convert a string into a LocalDateTime object.
+     * @param commandLine This is part of the user input.
+     * @return A LocalDateTime object, based on the user input.
+     */
+    static LocalDateTime parseDateTime(String commandLine) {
```

Comment:
I like how the individual components of the method body are attached with a comment. It improves readability.